core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -237,7 +239,7 @@ private void extract(PluginHandle pluginHandle, Terminal terminal, Path pluginFi
         Path root = tmp.resolve(pluginHandle.name);
         unzipPlugin(pluginFile, root);
 
-        // find the actual root (in case its unzipped with extra directory wrapping)
+        // find the actual root for the elasticsearch plugin (the zip may contain other system plugins).
         root = findPluginRoot(root);
```

Comment:
@rmuir is it ? I am not sure I understand what you mean by "not play well with this change at all". The change is minimal and should not impact anything. There is a new directory at the root of the plugin (named elasticsearch) that is checked in the PluginManager but the directory will not be present in the plugins directory.